cmd/promtool/main.go

```diff
@@ -1332,6 +1335,57 @@ func (j *jsonPrinter) printLabelValues(v model.LabelValues) {
 	json.NewEncoder(os.Stdout).Encode(v)
 }
 
+type unittestPrinter struct {
+	Step model.Duration
+}
+
+func (u *unittestPrinter) printValue(v model.Value) {
+	samples := make(map[string][]string)
+
+	switch modelType := v.(type) {
+	case model.Vector:
+
+		for _, samplePair := range modelType {
+			metricName := samplePair.Metric.String()
+
+			if _, ok := samples[metricName]; !ok {
+				samples[metricName] = []string{}
+			}
+
+			samples[metricName] = append(samples[metricName], strconv.FormatFloat(float64(samplePair.Value), 'f', -1, 64))
+		}
+
+	case model.Matrix:
+		for _, stream := range modelType {
+			metricName := stream.Metric.String()
+
+			if _, ok := samples[metricName]; !ok {
+				samples[metricName] = []string{}
+			}
+
+			for _, samplePair := range stream.Values {
+				samples[metricName] = append(samples[metricName], strconv.FormatFloat(float64(samplePair.Value), 'f', -1, 64))
```

Comment:
The query might return samplePair.Histogram != nil, which is a native histogram. I'd just `panic` in the MVP for now to indicate that this is unsupported unless you want to add the support. I'm not actually sure we have a formatter to get you the right output (see https://prometheus.io/docs/prometheus/latest/configuration/unit_testing_rules/#series).